alica_tests/src/test/test_alica_scheduling.cpp

```diff
@@ -224,5 +224,36 @@ TEST_F(AlicaSchedulingPlan, behaviourRunCheck)
     }
 }
 
+TEST_F(AlicaSchedulingPlan, execBehaviourCheck)
+{
+    CounterClass::called = -1;
+    ae->start();
+
+    auto& wm = alica_test::SchedWM::instance();
+    wm.execBehaviourTest = true;
+    ac->stepEngine();
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(50));
+    ASSERT_TRUE(wm.executeBehaviourRunCalled);
+
+    alica_test::SchedWM::instance().transitionToExecuteBehaviourInSubPlan = true;
+    std::this_thread::sleep_for(std::chrono::milliseconds(50));
```

Comment:
this won't work as expected. There is nothing stopping the run method in TestBehaviour (ExecuteBehaviour state) from executing when the transitionToExecuteBehaviourInSubPlan is true. Instead you need to ascertain that run/terminate is called when the transition is true & then init/run is called for the next state. Have a look at the BehAAA & other scheduling tests to get an idea of how to do this.